modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/saml/SAMLLocalAndOutboundAuthenticatorsTestCase.java

```diff
@@ -0,0 +1,652 @@
+        remoteUSMServiceClient = new RemoteUserStoreManagerServiceClient(backendURL, sessionCookie);
+        identityProviderMgtServiceClient = new IdentityProviderMgtServiceClient(sessionCookie, backendURL);
+        httpClient = new DefaultHttpClient();
+
+        createIDP();
+        createUser();
+        createLocalAndOutBoundAuthenticator();
+        createApplication();
+        ssoConfigServiceClient.addServiceProvider(getSsoServiceProviderDTO());
+
+        //Starting tomcat
+        log.info("Starting Tomcat");
+        tomcatServer = Utils.getTomcat(getClass());
+
+        URL resourceUrl = getClass().getResource(File.separator + "samples" + File.separator + config.getApplication
+                ().getArtifact() + "" + ".war");
+        Utils.startTomcat(tomcatServer, "/" + config.getApplication().getArtifact(), resourceUrl.getPath());
+
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testClear() throws Exception {
+        deleteUser();
+        deleteApplication();
+        deleteIDP();
+        ssoConfigServiceClient = null;
+        applicationManagementServiceClient = null;
+        remoteUSMServiceClient = null;
+        httpClient = null;
+        //Stopping tomcat
+        tomcatServer.stop();
+        tomcatServer.destroy();
+        resetISConfiguration();
+        Thread.sleep(10000);
+    }
+
+    @Test(description = "Test whether error code included in redirect URL with DefaultAuthenticator", groups = "wso2" +
+            ".is", priority = 1)
+    public void testErrorCodeInRedirectUrl() throws Exception {
+        try {
+            HttpResponse response;
+            String redirectUrl;
+            response = Utils.sendGetRequest(String.format(SAML_SSO_LOGIN_URL, config.getApplication().getArtifact(),
+                    config.getHttpBinding().binding), USER_AGENT, httpClient);
+
+            if (config.getHttpBinding() == HttpBinding.HTTP_POST) {
+                String samlRequest = Utils.extractDataFromResponse(response, "SAMLRequest", 5);
+                Map<String, String> paramters = new HashMap<String, String>();
+                paramters.put("SAMLRequest", samlRequest);
+                response = Utils.sendSAMLMessage(SAML_SSO_URL, paramters, USER_AGENT, config.getUserMode(),
+                        TENANT_DOMAIN_PARAM, config.getUser().getTenantDomain(), httpClient);
+                EntityUtils.consume(response.getEntity());
+                response = Utils.sendRedirectRequest(response, USER_AGENT, ACS_URL, config.getApplication()
+                        .getArtifact(), httpClient);
+            }
+            String sessionKey = Utils.extractDataFromResponse(response, "name=\"sessionDataKey\"", 1);
+            response = sendPostMessage(sessionKey);
+            EntityUtils.consume(response.getEntity());
+            redirectUrl = Utils.getRedirectUrl(response);
+            Assert.assertTrue(StringUtils.contains(redirectUrl,"errorCode"));
```

Comment:
Shall we specify the error code for each scenario in data provider and check them here?